scripts/data_prep/convert_delta_to_json.py

```diff
@@ -0,0 +1,451 @@
+
+    It can handle the current outptu formats that are supported by the server.
+    In contrast to the regular API methods of the client, this method does not
+    return the schema and drops all other responses.
+    """
+    req = self._execute_plan_request_with_metadata()
+    req.plan.CopyFrom(plan)
+
+    # Add the request options
+    if type == 'json':
+        format = cloud_pb2.ResultOptions.CloudOptions.FORMAT_JSON
+    elif type == 'csv':
+        format = cloud_pb2.ResultOptions.CloudOptions.FORMAT_CSV
+    elif type == 'arrow':
+        format = cloud_pb2.ResultOptions.CloudOptions.FORMAT_ARROW
+    else:
+        raise Exception('Invalid type')
+
+    ro = cloud_pb2.ResultOptions(
+        type=cloud_pb2.ResultOptions.TYPE_CLOUD,
+        cloudOptions=cloud_pb2.ResultOptions.CloudOptions(
+            format=format,
+            useCompression=False,
+        ))
+    cloud_option = any_pb2.Any()
+    cloud_option.Pack(ro)
+    req.request_options.append(
+        pb2.ExecutePlanRequest.RequestOption(extension=cloud_option))
+
+    # Create the iterator
+    iterator = ExecutePlanResponseReattachableIterator(req, self._stub,
+                                                       self._retry_policy,
+                                                       self._builder.metadata())
+    # Iterate over the response
+
+    result = []
+    row_count = 0
+    is_overflow = False
+
+    for response in iterator:
+        if response.HasField('extension') and response.extension.Is(
+                cloud_pb2.CloudResultBatch.DESCRIPTOR):
+            batch = cloud_pb2.CloudResultBatch()
+            assert response.extension.Is(cloud_pb2.CloudResultBatch.DESCRIPTOR)
+            response.extension.Unpack(batch)
+            result += [
+                Result(b.url, b.row_count, b.compressed_size,
+                       b.uncompressed_size) for b in batch.results
+            ]
+            row_count += sum(result.row_count for result in batch.results)
+            is_overflow |= batch.truncated
+    return result, row_count, is_overflow
+
+
+SparkConnectClient.to_cf = to_cf  # pyright: ignore
+
+
+def collect_as_cf(self: DataFrame,
+                  type: str = 'json') -> Tuple[List[Result], int, bool]:
+    query = self._plan.to_proto(self._session.client)  # pyright: ignore
+    return self._session.client.to_cf(query, type)  # pyright: ignore
+
+
+DataFrame.collect_cf = collect_as_cf  # pyright: ignore
+
+
+def iterative_combine_jsons(json_directory: str, output_file: str) -> None:
+    """Combine jsonl files in json_directory into one big jsonl file.
+
+    Args:
+        json_directory(str): directory containing the JSON files
+        output_file(str): output JSONL file
+    """
+    json_files = [f for f in os.listdir(json_directory) if f.endswith('.jsonl')]
+    with open(output_file, 'w') as outfile:
+        for file_name in json_files:
+            with open(os.path.join(json_directory, file_name), 'r') as infile:
```

Comment:
this seems like it is written to merge many jsonl files into one jsonl file, but the docstring says it is intended to convert many json files into one jsonl files. Is the implementation wrong or the docstring wrong (or am I missing something)?

Reply:
it is merging many jsonl files into one jsonl. docstring is updated.